ENIGMAsystem/SHELL/Platforms/General/POSIX/main.cpp

```diff
@@ -40,6 +42,25 @@ void initialize_directory_globals() {
 
 }
 
+namespace enigma_user {
+
+string shellscript_evaluate(string command, size_t buffer_size) {
```

Comment:
Passing in a `buffer_size` is an unnecessary implementation detail. This function doesn't need one to work (and work correctly).

Also, the return type does not give the user a way to check whether the call succeeded. Usually, if a user cares about data that was printed, they care about whether the routine succeeded as well.

Reply:
Again, how would I go about doing that? I have no idea.

Reply:
Explained above.